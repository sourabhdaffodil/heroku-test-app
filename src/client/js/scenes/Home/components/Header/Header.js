import React from 'react';
import styled from 'styled-components';
import { media } from '../../../../theme/style-utils';
import RestricitedContainer from '../../../../components/Containers/Restricited/Restricited';
import SpaceBwContainer from '../../../../components/Containers/SpaceBetween/SpaceBetween';
import Link from '../../../../components/Link/Link';
import Social from '../Social/Social';

const HeaderWrapper = styled.header`
  background-color: ${props => props.theme.color('foreground')};
  padding: 12px 0;
  color: ${props => props.theme.contrast('dark')};
  ${media.phone`display: none;`}
`;

const Tag = styled.span`
  color: ${props => props.theme.contrast('dark', 'secondary')};
  font-weight: ${props => props.theme.weight('semiBold')};
  ${props => props.theme.textSetting('nr')};
`;

const Header = () => (
  <HeaderWrapper>
    <RestricitedContainer>
      <SpaceBwContainer>
        <Link url="https://twitter.com/orenciorodolfo">
          <Tag>@orenciogoncalves</Tag>
        </Link>
        <Social />
      </SpaceBwContainer>
    </RestricitedContainer>
  </HeaderWrapper>
);

export default Header;
