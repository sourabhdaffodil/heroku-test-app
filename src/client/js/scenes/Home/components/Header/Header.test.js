import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header component', () => {
  it('Must render the left content of a pill button', () => {
    const wrapper = shallow(
      <Header>
        Left content
      </Header>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
