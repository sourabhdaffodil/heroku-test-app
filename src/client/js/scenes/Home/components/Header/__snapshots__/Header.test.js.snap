// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Must render the left content of a pill button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header>
    Left content
  </Header>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <Container>
          <Link
            className=""
            url="https://twitter.com/orenciorodolfo"
          >
            <styled.span>
              @orenciogoncalves
            </styled.span>
          </Link>
          <Connect(Social) />
        </Container>
      </Container>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": <Container>
          <Link
            className=""
            url="https://twitter.com/orenciorodolfo"
          >
            <styled.span>
              @orenciogoncalves
            </styled.span>
          </Link>
          <Connect(Social) />
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              className=""
              url="https://twitter.com/orenciorodolfo"
            >
              <styled.span>
                @orenciogoncalves
              </styled.span>
            </Link>,
            <Connect(Social) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.span>
                @orenciogoncalves
              </styled.span>,
              "className": "",
              "url": "https://twitter.com/orenciorodolfo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "@orenciogoncalves",
              },
              "ref": null,
              "rendered": "@orenciogoncalves",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <Container>
            <Link
              className=""
              url="https://twitter.com/orenciorodolfo"
            >
              <styled.span>
                @orenciogoncalves
              </styled.span>
            </Link>
            <Connect(Social) />
          </Container>
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <Container>
            <Link
              className=""
              url="https://twitter.com/orenciorodolfo"
            >
              <styled.span>
                @orenciogoncalves
              </styled.span>
            </Link>
            <Connect(Social) />
          </Container>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Link
                className=""
                url="https://twitter.com/orenciorodolfo"
              >
                <styled.span>
                  @orenciogoncalves
                </styled.span>
              </Link>,
              <Connect(Social) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": <styled.span>
                  @orenciogoncalves
                </styled.span>,
                "className": "",
                "url": "https://twitter.com/orenciorodolfo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "@orenciogoncalves",
                },
                "ref": null,
                "rendered": "@orenciogoncalves",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
